src/core/tranche/distributor/maker.sol

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+import "../distributor.sol";
+import "../../lightswitch.sol";
+
+contract MakerDistributor is Distributor, Switchable {
+
+    function balance() public auth {
+
+//        require(flowThrough==false);
+
+        if (manager.poolClosing()) {
+            repayTranches();
```

Comment:
I am a bit confused. Are we deciding if we repay here with a flag or with the return of pileCheck?
Currently, it is both.